components/ntp_sponsored_images/locale_helper_linux.cc

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <stdlib.h>
+
+#include "brave/components/ntp_sponsored_images/locale_helper_linux.h"
+
+namespace internal {
+
+LocaleHelperLinux::LocaleHelperLinux() = default;
+
+LocaleHelperLinux::~LocaleHelperLinux() = default;
+
+const std::string LocaleHelperLinux::GetLocale() const {
+  char const *language = nullptr;
+
+  if (!language || !*language) {
```

Comment:
Curious what rewards/ads is doing for locale detection? (would they be able to make use of this too?)

As a bonus (for reference), I wanted to share the implementation from a previous life when I did this for Muon:
https://github.com/brave/muon/pull/647/files

Reply:
Oh, you already did same job for Muon 👍

I just copied the code used by ads for lang/region detection.
Of course, we can just call that API by adding brave_ads dependency to this module instead of copy. The reason why I copied was dependency issue. brave_ads component has some dependency to brave layer. It's illegal now. Because of that, this module also gets that illegal dependency.

I think we could make this local helper logic as a separate module and can share it with this and other modules such as brave_ads as a f/u.